sky/shell/shell.cc

```diff
@@ -68,6 +73,81 @@ void ServiceIsolateHook(bool running_precompiled) {
   }
 }
 
+class JSONResponseBuilder {
+ public:
+  void Append(const char* str) {
+    for (intptr_t i = 0; str[i] != '\0'; i++) {
+      output_.push_back(str[i]);
+    }
+  }
+
+  void Append(const std::string& str) {
+    for (size_t i = 0; i < str.size(); i++) {
+      output_.push_back(str.data()[i]);
+    }
+  }
+
+  void Append(uintptr_t num) {
+    Append(std::to_string(num));
+  }
+
+  std::string ToString() {
+    return std::string(output_.data(), output_.size());
+  }
+
+ private:
+  std::vector<char> output_;
+};
+
+// This callback will be invoked on a Dart worker thread.
+bool FlutterRunInView(const char* method,
```

Comment:
We should move these functions to a different file and reference them from this file.  Files like shell.cc tend to accumulate lots of code because they're central hubs, so its worth trying to push as much code out of them as possible.